In [4]:
from sprt.sequential_test_monte_carlo import sequential_test_simulation
from sprt.ab_stat_tests import calc_prop_test_sample_size

In [14]:
COVERSION_RATE = 0.05
MDE = 0.1
MDE_REAL = 0.1
ALPHA = 0.1
POWER = 0.9
N_SIMULATIONS = 10_000
SIMULATION_STEP = 1000 # how often do we check the result of the test

In [15]:
print('z-test sample size per variation:', 
      calc_prop_test_sample_size(cr=COVERSION_RATE, mde=MDE, power=POWER, alpha=ALPHA, alternative='greater'))

z-test sample size per variation: 24964


In [16]:
sequential_test_simulation(COVERSION_RATE, 
                           MDE, 
                           MDE_REAL, 
                           ALPHA, 
                           POWER, 
                           N_SIMULATIONS, 
                           SIMULATION_STEP)


Type I Error 0.07
Type II Error 0.14
-----------------------------------
Stopped earlier if H0 True 0.18
Stopped earlier if H1 True 0.47
-----------------------------------
Saved time if H0 True 0.06
Saved time if H1 True 0.15


| Parameters  | Stopped earlier | Saved time | Sample size per var |
| --- | --- | --- | --- |
| CR 15% |
| MDE=5%, MDE_REAL=5% | 85% / 97% | 47% / 59% | 30k |
| MDE=10%, MDE_REAL=10% | 23% / 68% | 8% / 25% | 7k |
| MDE=15%, MDE_REAL=15% | 3% / 35% | 1% / 7% | 3k |
| MDE=5%, MDE_REAL=10% | 85% / 99% | 47% / 75% | 
| MDE=5%, MDE_REAL=-5% | 85% / 99% | 48% / 80% | 
| CR 35% |
| MDE=5%, MDE_REAL=5% | 86% / 81% | 50% / 46% | 10k |
| MDE=10%, MDE_REAL=10% | 21% / 66% | 7% / 24% | 3k |
| MDE=15%, MDE_REAL=15% | 2% / 30% | 1% / 7% | 1k |
| MDE=5%, MDE_REAL=10% | 85% / 79% | 50% / 49% | 